addons/binding/org.openhab.binding.chamberlainmyq/src/main/java/org/openhab/binding/chamberlainmyq/handler/ChamberlainMyQGatewayHandler.java

```diff
@@ -0,0 +1,472 @@
+
+import static org.openhab.binding.chamberlainmyq.ChamberlainMyQBindingConstants.*;
+
+import java.io.IOException;
+import java.io.UnsupportedEncodingException;
+import java.net.URLEncoder;
+import java.util.List;
+import java.util.Properties;
+import java.util.concurrent.Executors;
+import java.util.concurrent.ScheduledExecutorService;
+import java.util.concurrent.ScheduledFuture;
+import java.util.concurrent.TimeUnit;
+
+import org.apache.commons.io.IOUtils;
+import org.apache.commons.lang.StringUtils;
+import org.eclipse.smarthome.config.discovery.DiscoveryService;
+import org.eclipse.smarthome.core.thing.Bridge;
+import org.eclipse.smarthome.core.thing.ChannelUID;
+import org.eclipse.smarthome.core.thing.Thing;
+import org.eclipse.smarthome.core.thing.ThingStatus;
+import org.eclipse.smarthome.core.thing.ThingStatusDetail;
+import org.eclipse.smarthome.core.thing.binding.BaseBridgeHandler;
+import org.eclipse.smarthome.core.types.Command;
+import org.openhab.binding.chamberlainmyq.config.ChamberlainMyQGatewayConfig;
+import org.openhab.binding.chamberlainmyq.internal.ChamberlainMyQResponseCode;
+import org.openhab.binding.chamberlainmyq.internal.HttpUtil;
+import org.openhab.binding.chamberlainmyq.internal.InvalidLoginException;
+import org.openhab.binding.chamberlainmyq.internal.discovery.ChamberlainMyQDeviceDiscoveryService;
+import org.openhab.binding.chamberlainmyq.internal.json.Device;
+import org.openhab.binding.chamberlainmyq.internal.json.MyqJson;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import com.google.gson.Gson;
+import com.google.gson.JsonObject;
+import com.google.gson.JsonParser;
+
+//import org.eclipse.smarthome.io.net.http.HttpUtil;
+
+/**
+ * The {@link ChamberlainMyQGatewayHandler} is responsible for handling commands, which are
+ * sent to one of the channels.
+ *
+ * @author Scott Hanson - Initial contribution
+ */
+public class ChamberlainMyQGatewayHandler extends BaseBridgeHandler {
+
+    private Logger logger = LoggerFactory.getLogger(ChamberlainMyQGatewayHandler.class);
+
+    private String securityToken;
+    ScheduledFuture<?> mainPollRefreshJob;
+    private int refreshInterval;
+    private int quickPoll;
+
+    private ScheduledExecutorService pollService = Executors.newSingleThreadScheduledExecutor();
+
+    /**
+     * The regular polling task
+     */
+    private ScheduledFuture<?> pollFuture;
+
+    /**
+     * This task will reset the poll interval back to normal after a rapid poll
+     * cycle
+     */
+    private ScheduledFuture<?> pollResetFuture;
+
+    /**
+     * Cap the time we poll rapidly to not overwhelm the servers with api
+     * requests.
+     */
+    private static int MAX_RAPID_REFRESH = 30 * 1000;
+    // private Properties header;
+
+    private final Gson gson = new Gson();
+
+    public ChamberlainMyQGatewayHandler(Bridge bridge) {
+        super(bridge);
+    }
+
+    @Override
+    public void handleCommand(ChannelUID channelUID, Command command) {
```

Comment:
All bindings should aim to implement at least the REFRESH command, is there a reason not do that here?